test/controllers/ApplicationControllerSpec.scala

```diff
@@ -0,0 +1,43 @@
+package controllers
+
+import config.CSRHttp
+import connectors.ApplicationClient
+
+class ApplicationControllerSpec extends BaseControllerSpec {
+  val applicationClient = mock[ApplicationClient]
+
+  class TestableApplicationController extends ApplicationController(applicationClient) with TestableSecureActions {
+    val http: CSRHttp = CSRHttp
+    override protected def env = securityEnvironment
+  }
+
+  def controller = new TestableApplicationController
+
+  "index" should {
```

Comment:
Don't seem to have added any tests for the cookie stuff. Is that because it is tricky to test?